build.sbt

```diff
@@ -0,0 +1,116 @@
+  Seq() ++ {
+    // If we're building with Scala > 2.11, enable the compile option
+    //  switch to support our anonymous Bundle definitions:
+    //  https://github.com/scala/bug/issues/10047
+    CrossVersion.partialVersion(scalaVersion) match {
+      case Some((2, scalaMajor: Long)) if scalaMajor < 12 => Seq()
+      case _ => Seq("-Xsource:2.11")
+    }
+  }
+}
+
+def javacOptionsVersion(scalaVersion: String): Seq[String] = {
+  Seq() ++ {
+    // Scala 2.12 requires Java 8. We continue to generate
+    //  Java 7 compatible code for Scala 2.11
+    //  for compatibility with old clients.
+    CrossVersion.partialVersion(scalaVersion) match {
+      case Some((2, scalaMajor: Long)) if scalaMajor < 12 =>
+        Seq("-source", "1.7", "-target", "1.7")
+      case _ =>
+        Seq("-source", "1.8", "-target", "1.8")
+    }
+  }
+}
+
+val defaultVersions = Map("chisel3" -> "latest.release")
+
+lazy val commonSettings = Seq (
+  organization := "edu.berkeley.cs",
+  version := "2.0",
+//  git.remoteRepo := "git@github.com:ucb-bar/riscv-sodor.git",
+  autoAPIMappings := true,
+  scalaVersion := "2.11.12",
+  crossScalaVersions := Seq("2.11.12", "2.12.6"),
+  scalacOptions := Seq("-deprecation", "-feature") ++ scalacOptionsVersion(scalaVersion.value),
+  javacOptions ++= javacOptionsVersion(scalaVersion.value),
+  libraryDependencies ++= Seq(
+    "com.github.scopt" %% "scopt" % "3.7.0"
+  ),
+  libraryDependencies ++= (Seq("chisel3").map {
+    dep: String => "edu.berkeley.cs" %% dep % sys.props.getOrElse(dep + "Version", defaultVersions(dep))
+  })
+)
+
+lazy val common = (project in file("src/common")).
+  settings(commonSettings: _*)
+
+lazy val rv32_1stage = (project in file("src/rv32_1stage")).
+  settings(commonSettings: _*).
+  settings(chipSettings: _*).
+  dependsOn(common)
+
+lazy val rv32_2stage = (project in file("src/rv32_2stage")).
+  settings(commonSettings: _*).
+  settings(chipSettings: _*).
+  dependsOn(common)
+
+lazy val rv32_3stage = (project in file("src/rv32_3stage")).
+  settings(commonSettings: _*).
+  settings(chipSettings: _*).
+  dependsOn(common)
+
+lazy val rv32_5stage = (project in file("src/rv32_5stage")).
+  settings(commonSettings: _*).
+  settings(chipSettings: _*).
+  dependsOn(common)
+
+
+lazy val rv32_ucode  = (project in file("src/rv32_ucode")).
+  settings(commonSettings: _*).
+  settings(chipSettings: _*).
+  dependsOn(common)
+
+
+lazy val elaborateTask = InputKey[Unit]("elaborate", "convert chisel components into backend source code")
+lazy val makeTask = InputKey[Unit]("make", "trigger backend-specific makefile command")
+
+def runChisel(args: Seq[String], cp: Classpath, pr: ResolvedProject) = {
+   val numArgs = 1
+   require(args.length >= numArgs, "syntax: elaborate <component> [chisel args]")
+   val projectName = pr.id
+//   val packageName = projectName //TODO: valid convention?
+   val packageName = "Sodor" //TODO: celio change
```

Comment:
Are these TODOs still valid?

Reply:
No clue; I don't understand build.sbt conventions and I don't remember the history behind that comment.